master/internal/api_experiment.go

```diff
@@ -46,6 +46,60 @@ import (
 
 var experimentsAddr = actor.Addr("experiments")
 
+// Catches information on active running trials.
+type experimentAllocation struct {
+	JobID       *string
+	NumRunning  int32
+	NumStarting int32
+	TaskID      *string
+}
+
+func (a *apiServer) enrichExperimentState(experiments []*experimentv1.Experiment) (
+	[]*experimentv1.Experiment, error,
+) {
+	// filter allocations by JobIDs on this page of experiments
+	jobFilter := []string{}
+	for _, exp := range experiments {
+		jobFilter = append(jobFilter, exp.JobId)
+	}
+
+	// get active or pending tasks
+	tasks := []*experimentAllocation{}
```

Comment:
use a specialized struct for each query, maybe use bun if you are brave.

Reply:
I made separate taskAllocation and experimentAllocation, is that what you meant?